File: Randoms/350. Intersection of Two Arrays II.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        #using hashmap
        count = {} #hashmap
        res= []

        #for counting frequency
        for n in nums1:
            if n in count:
                count[n]+=1
            else:
                count[n]=1
            #or: count[n]=count.get(n,0)+1
        
        for n in nums2:
            if n in count and count[n]>0:
                res.append(n)
                count[n]-=1
                #decreasing count to avoid using the same element more times than it actually exists
        return res

File: Randoms/350. Intersection of Two Arrays II.ts
function intersect(nums1: number[], nums2: number[]): number[] {
    const count: Record<number, number>={}; //hashmap to record frequency
    const res = [] //empty array

    for (const n of nums1){
        if(n in count){
            count[n]+=1
        }
        else{
            count[n]=1
        }
    }

    for (const n of nums2){
        if(n in count && count[n]>0){
            res.push(n);
            count[n]-=1;
        }
    }

    return res
};
